packages/tectonic-explorer/js/components/draggable-dialog.tsx

```diff
@@ -37,7 +37,9 @@ interface IProps {
 
 // patterned after https://mui.com/components/dialogs/#draggable-dialog
 export const DraggableDialog: React.FC<IProps> = ({ backdrop, onClose, title, offset, initialPosition, children }) => {
-  function PaperComponent(props: PaperProps) {
+  // PaperComponent needs to be memoized, as otherwise it's recreated on every render. This might lead to unexpected
+  // side effects like losing input focus on each render.
+  const PaperComponentMemoized = useMemo(() => function PaperComponent(props: PaperProps) {
```

Comment:
You could also move the definition of `PaperComponent` outside `DraggableDialog` altogether and pass in the `offset` as a prop (which I think is the only additional context required) which would avoid the re-creation of the component on every render (which is still occurring--the `useMemo()` is just preventing downstream code from noticing 😉) and eliminate the need for `useMemo()`.

Reply:
I was hoping I could do that, but note that `PaperComponent` is later passed to another component as property with constructor/class value, not even JSX element. I cannot modify its props and add `offset` there, as I don't have control over what its parent will pass to it. It'd need to be something like HOC, but we'd be back to the same problem I think.